airbyte-integrations/connectors/source-github/source_github/streams.py

```diff
@@ -337,6 +347,7 @@ def read_records(
         ):
             cursor_value = self.convert_cursor_value(record[self.cursor_field])
             if not start_point or cursor_value > start_point:
+                self.state = self._get_updated_state(self.state, record)
```

Comment:
This works under the condition that we don't use `self._state` in `stream_slices` because of the availability strategy. This seems a bit dangerous but I don't have a better way to do that...

Reply:
This also assumes that the records read during the availability check are older than the first one synced in the read_records. This also seems like a fair assumption but the flow is starting to get very complex it feels

Reply:
Noting here that the call to update state has been shifted to occur AFTER a record is yielded, to ensure state is only updated after a record has been processed.